src/tasks/LambdaDeployFunction/TaskOperations.ts

```diff
@@ -66,11 +69,16 @@ export class TaskOperations {
             }
 
             const response = await this.lambdaClient.updateFunctionCode(updateCodeRequest).promise()
-
             if (!response.FunctionArn) {
                 throw new Error(tl.loc('NoFunctionArnReturned'))
             }
 
+            console.log(tl.loc('AwaitingStatus', this.taskParameters.functionName, FUNCTION_UPDATED))
+            await this.lambdaClient
+                .waitFor(FUNCTION_UPDATED, { FunctionName: this.taskParameters.functionName })
+                .promise()
+            console.log(tl.loc('AwaitingStatusComplete', this.taskParameters.functionName, FUNCTION_UPDATED))
+
```

Comment:
Turn into method `waitForUpdate` ?

Reply:
Could do one for `FUNCTION_UPDATED` but Typescript doesn't like mixing and matching the args for different overloads, so I wouldn't do one for `FUNCTION_ACTIVE`. Would feel kinda silly to make two helper functions calling the same thing in the same way just for the sole purpose of making Typescript happy.

![image](https://user-images.githubusercontent.com/29374703/141829578-3ee81305-a8cd-44d5-9e0c-f8d007d8c246.png)

Reply:
Ah okay. I need to start reviewing PRs in an editor so I can look at these things.
This is a known Typescript issue and boils down to how overloads work: https://github.com/microsoft/TypeScript/issues/14107 (in summary: avoid overloads if possible)

So yeah it's good as-is.